app/Services/ModuleService.php

```diff
@@ -70,4 +81,139 @@ public function getAdminLinks(): array
     {
         return self::$adminLinks;
     }
+
+    public function createModule($array): bool
+    {
+        $orig_file = $array['file'];
+        $file_ext = $orig_file->getClientOriginalExtension();
+        $allowed_extensions = ['zip', 'tar', 'gz'];
+
+        if (!in_array($file_ext, $allowed_extensions)) {
+            return false;
+        }
+
+        $module = null;
+        $temp = storage_path('app/tmp');
```

Comment:
You should create a subdirectory for the module and extract it all into there. They're getting extracted into the root of this directory, it looks like. If there are 2 modules uploaded the files could mix, if it isn't properly cleaned up (or wasn't cleaned up, for some reason). Also, there's a `.gitignore` file in this which probably shouldn't be getting copied over. So a subdirectory for the module would be ideal. And then when it's extract, you need to check the extracted module got extracted into an additional subdirectory, or into the root of the directory we created - because the module creator might have either a) added the subdirectory (e.g, `my-module` into the archive file or b) just created the archive file at the root of the module instead of including the folder itself. 

Does that make sense? Just 2 use-cases here that need to be covered

Reply:
Yeah thanks! Did the following:
Made it extract to temporary unique folder inside app/tmp/modules/ and then do the same as coded. 
By the way sorry for closing the pull request. It happened accidentally while browsing github app :p